src/word-count/index.jsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState, useCallback } from 'react';
+import { TextArea } from '@ukhomeoffice/react-components';
+import classNames from 'classnames';
+
+export default function WordCount(props) {
+
+    const { value, maxWordCount, error, values, name } = props;
+    const getWordCount = text => text?.split(/\s+/).filter(Boolean).length;
+
+    const [{ content, wordCount }, setContent] = useState({
+        content: value,
+        wordCount: getWordCount(value) ?? 0,
+    });
+
+    const handleChange = useCallback(text => {
+
+        const wordCount = getWordCount(text);
+
+        if (wordCount > maxWordCount) {
+            setContent({
+                content: text,
+                wordCount
+            });
+        } else {
+            setContent({
+                content: text,
+                wordCount
+            });
+        }
+    },[maxWordCount, content]);
+
+    const wordCountHintMessage = wordCount => {
```

Comment:
If you make this a component react will better be able to cache/reuse it on rerenders where the wordcount doesn't change.

```suggestion
    const WordCountHintMessage = ({wordCount}) => {
```

then call with `<WordCountHintMessage {...(wordCount}} />`